ns1/ipam.py

```diff
@@ -0,0 +1,446 @@
+    def new_address(self, prefix, type, callback=None, errback=None):
+        """
+        Create a new address space in this VRF
+        :param str prefix: The CIDR prefix of the address to add
+        :param str type: planned, assignment, host
+        :param callback:
+        :param errback:
+        :return:
+        """
+        if not self.data:
+            raise VrfException('VRF not loaded')
+
+        return Address(self.config, prefix, type, self).create()
+
+
+class Address(object):
+
+    def __init__(self, config, prefix=None, type=None, vrf=None, id=None):
+        """
+        Create a new high level Address object
+
+        :param ns1.config.Config config: config object
+        :param str prefix: cidr prefix
+        :param str type: planned, assignment, host
+        :param Vrf vrf: VRF Object the address will be part of
+        """
+        self._rest = Addresses(config)
+        self.config = config
+        self.prefix = prefix
+        self.type = type
+        self.vrf = vrf
+        self.id = id
+        self.children = []
+        self.report = {}
+        self.data = None
+
+    def __repr__(self):
+        return '<Address address=%s>' % self.prefix
+
+    def __getitem__(self, item):
+        return self.data.get(item, None)
+
+    def reload(self, callback=None, errback=None):
+        """
+        Reload address data from the API.
+        """
+        return self.load(reload=True, callback=callback, errback=errback)
+
+    def load(self, callback=None, errback=None, reload=False):
+        """
+        Load address data from the API.
+        """
+        if not reload and self.data:
+            raise AddressException('Address already loaded')
+
+        def success(result, *args):
+            self.data = result
+            self.id = result.id
+            self.prefix = result['prefix']
+            self.type = result['type']
+            self.vrf = Vrf(self.config, id=result['vrf_id'])
+            if self.type != 'host':
+                self.report = self._rest.report(self.id)
+                children = self._rest.retrieve_children(self.id)
+                self.children = [Address(self.config, id=child['id']) for child in
+                                 children if len(children) > 0]
+            try:
+                parent = self._rest.retrieve_parent(self.id)
+                self.parent = Address(self.config, id=parent['id'])
+            except ResourceException:
+                pass
+            if callback:
+                return callback(self)
+            else:
+                return self
+
+        if self.id is None:
+            if self.prefix is None or self.type is None or self.vrf is None:
+                raise AddressException('Must at least specify an id or prefix, type and vrf')
+            else:
+                vrf_id = self.vrf.id
+                try:
+                    self.id = [address for address in self._rest.list() if address['prefix'] == self.prefix and address[
```

Comment:
This should be available as an API call soon if not yet already.

Reply:
You referring to search?

Reply:
Oh ya. Used to be called find by prefix.